### DISCUSSION TITLE
Can I apply a trained with exogenous variables Nihits model to new time series?

### DISCUSSION THREAD

**Discussion by tana777** [Nixtla/neuralforecast#719] [Q&A]
Hi, I'm following this [document](https://nixtla.github.io/neuralforecast/examples/exogenous_variables.html) to train a Nhits model with exogenous variables. It went well but when I used the model to make forecasts on unseen time series, I got all empty results. So is it possible to apply it to new time series? If not, what stops the model from doing that? If it's possible, am I missing anything? Thank you and let me know if you need any details.

**Comment by cchallu** [comment]
Hi @tana777! Sorry for the late reply. Yes, it is possible. We just released some fixes that will provide more feedback when there is an issue with exogenous variables.

The only requirement for it to work is that all exogenous variables listed in the 3 types of exogenous variables (`stat`, `futr`, and `hist`) are always present in the corresponding dataset. Static variables should be included in the `static_df`. `hist` and `futr` variables in the historic dataset (`df` argument). And finally for the `predict` method, also add the `futr` variables in the `futr_df` dataset.

I hope this is useful. I can provide a more detailed answer if you show the error and parts of your code (if it's not confidential).
